source-only mode: add initial document (#527)
fluent/fluentd-docs-gitbook: deployment/source-only-mode.md

# Source Only Mode

Since v1.18.0, Fluentd can launch with source-only mode.
(Not supported on Windows)

> * In this mode, only input plugins runs.

Reviewer: ```suggestion
* In this mode, only input plugins run.
```

This is because input plugins are not 3rd person singular.

* During this mode, the events are stored in a temporary file buffer.
* Sending `SIGWINCH` to the supervisor cancels this mode.
  * Then, all plugins start to run, and the temporary file buffer starts to load.

## How to launch Fluentd with source-only mode

You can launch Fluentd with source-only mode in the following ways.

* [Command Line Option - --with-source-only](command-line-option.md)
* [System Configuration - with_source_only](system-config.md#with_source_only)

## Temporary file buffer

> During source-only mode, the events are stored in a temporary file buffer.

Reviewer: What does the events actually mean?

The events should be ingested and stored. So, I'd like to recommend to describe more concretely like:

```suggestion
During source-only mode, the ingested events are stored in a temporary file buffer.
```

After `SIGWINCH` is sent to the supervisor and this mode is canceled, this buffer begins to load.

By default, the file buffer path is as follows.

> * `/tmp/fluent/source-only-buffer/{Unique ID for the Fluentd instance}`

Reviewer 2: It might be better to clarify "What instance".
e.g supervisor, worker,  plugin and so on.

Author: It is a common ID for the supervisor and all workers belonging to the supervisor, but I can't explain it briefly...

Reviewer 2: Hmm, so let's keep it as is.

You can confirm it in the following log output at startup.

```
[info]: #0 with-source-only: the emitted data will be stored in the buffer files under
/tmp/fluent/source-only-buffer/bbd9006d-bc41-418b-b346-f80888641dda. You can send SIGWINCH to
the supervisor process to cancel with-source-only mode and process data.
```

This file buffer is the buffer of [out_buffer](../output/buffer.md).
It works with the default settings of `out_buffer` except for the following points:

* `path` is automatically determined by default.
* `overflow_action` is `drop_oldest_chunk` by default.

> If need, you can configure some options for the buffer in System Configuration.

Reviewer: Just following the sentence structure, we need to change as follows:

```suggestion
If needed, you can configure some options for the buffer in System Configuration.
```

Please see the following for details.

> * [System Configuration - \<source_only_buffer\> section](system-config.md#less-than-source_only_buffer-greater-than-section).

Reviewer 3: Looks for me that the `less-than-source_only_buffer-greater-than-section` anchor is missing.
So, the browser cannot navigate to the appropriate part of the page...

Reviewer 3: When I preview this in local environment, it shows as following

![image](https://github.com/user-attachments/assets/2da8048e-1829-4f0c-834e-f25afed2fb02)

Seems the unexpected message is shown as link.

Author: Thanks, but behavior may be different from the production environment.

At least, the anchor would work because the following works now.

* https://docs.fluentd.org/output/file#less-than-buffer-greater-than-section
* https://github.com/fluent/fluentd-docs-gitbook/blob/2483910bbf16c723d052b38d92b7e43e2bfce2e7/output/file.md#L28

I cannot be sure about the broken description because there is no other similar section.
I will remove the `<` and `>` just in case.

Author: Fixed: de81205d8661b7d8e1a3ac738c39863536a68968

### Recovering

> If Fluentd stops with the temporary buffer left, you need to recover the buffer to launch Fluentd with source-only mode again.

Reviewer 3: ```suggestion
If Fluentd stops with the temporary buffer in place, you need to recover the buffer to launch Fluentd with source-only mode again.
```

Reviewer 2: or `remained`?

Author: Fixed! I use `remained`!

Note that a different path will be used each time unless you configure the temporary buffer path explicitly.
In this case, you can recover the buffer as follows.

> 1. Configure the left buffer path explicitly.

Reviewer 3: ```suggestion
1. Configure the remaining buffer path explicitly.
```

Author: Do you mean this is confusing with right/left?
Certainly, it is.
I will fix this.

Author: Fixed: de81205d8661b7d8e1a3ac738c39863536a68968

1. Start Fluentd with source-only mode again.
1. Send `SIGWINCH` to the supervisor to load the buffer.

> If this recovering is necessary, i.e., Fluentd stops with the temporary buffer left, the following warning log will be output.

Reviewer 3: ```suggestion
If this recovery is necessary, i.e., Fluentd stops with the temporary buffer in place, the following warning log will be output.
```

Author: Thanks, I will fix `recovering` to `recovery` entirely. => Fixed: de81205d8661b7d8e1a3ac738c39863536a68968

What is the problem with the following expression?

* `with the temporary buffer left`

Author: > What is the problem with the following expression?
> 
>     * `with the temporary buffer left`

Fixed! I use `remained` instead of `left`.

> If this recovering is necessary, i.e., Fluentd stops with the temporary buffer left, the following warning log will be output.

Reviewer: Including @Watson1978 's suggestion, I'm also recommend to use as follows here:

```suggestion
If this recovery is necessary, i.e., Fluentd stops with the temporary buffer left, the following warning log will be displayed.
```

Reviewer: Output is also able to use as verb in this case. But this type of usages is not common. So, I recommend to use display as a verb here. In this context, display should be changed its form as a past participle.
